keps/sig-network/1453-ingress-api/README.md

```diff
@@ -0,0 +1,1372 @@
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/20190731-production-readiness-review-process.md
+
+Production readiness review questionnaire must be completed for features in
+v1.19 or later, but is non-blocking at this time. That is, approval is not
+required in order to be in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+
+-->
+
+### Feature enablement and rollback
+
+<!-- _This section must be completed when targeting alpha to a release._ -->
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [ ] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name:
+    - Components depending on the feature gate:
+  - [x] Other
+    - Describe the mechanism: 
+        - `runtime-config ingress/v1,ingressclass/v1`
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+        - Yes, a restart of the apiserver and Ingress controllers is required if changed
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+        - No
+
+<!-- * **Does enabling the feature change any default behavior?** -->
+<!--   Any change of default behavior may be surprising to users or break existing -->
+<!--   automations, so be extremely careful here. -->
+
+<!-- * **Can the feature be disabled once it has been enabled (i.e. can we rollback -->
+<!--   the enablement)?** -->
+<!--   Also set `disable-supported` to `true` or `false` in `kep.yaml`. -->
+<!--   Describe the consequences on existing workloads (e.g. if this is runtime -->
+<!--   feature, can it break the existing applications?). -->
+
+<!-- * **What happens if we reenable the feature if it was previously rolled back?** -->
+
+<!-- * **Are there any tests for feature enablement/disablement?** -->
+<!--   The e2e framework does not currently support enabling and disabling feature -->
+<!--   gates. However, unit tests in each component dealing with managing data created -->
+<!--   with and without the feature are necessary. At the very least, think about -->
+<!--   conversion tests if API types are being modified. -->
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Try to be as paranoid as possible - e.g. what if some components will restart
+  in the middle of rollout?
```

Comment:
This is great now - thanks for this detailed answer.
FYI: I'm going to update the suggestions for the question a bit and will use generalized versions of those.

Reply:
Cool that these answers are useful